### PR TITLE
#184347042 Implement payment with stripe

### PR DESCRIPTION
**Pull request by irakozetony**
### What does this PR do?
Implement payment

### Description of Task to be completed?
User can be able to pay with their credits card, using the stripe api. When payment is completed, an order is created for the customer and a sales is created for the sellers

### How should this be manually tested?
1. Checkout to branch `feature/payment-184347042
2. Login as a buyer and add a few items
3. Send a post request to the endpoint `http://localhost:<5000>/api/v1/payment/create-checkout-session`
4. Then, visit the url returned in the response and enter card details to complete payment

#### What are the relevant pivotal tracker stories?
[#184347042](https://www.pivotaltracker.com/story/show/184347042)

### CODE REVIEW

**Review by ndatumuremyi — APPROVED** [review]

**Review by fabmukunzi — APPROVED** [review]

File: src/database/models/sale.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+const { Model } = require('sequelize');
+module.exports = (sequelize, DataTypes) => {
+	class Sales extends Model {
+		/**
+		 * Helper method for defining associations.
+		 * This method is not a part of Sequelize lifecycle.
+		 * The `models/index` file will call this method automatically.
+		 */
```
Comment (ncutixavier) [inline src/database/models/sale.js]: remove all unnecessary comment

File: src/routes/api/payment.routes.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	// restrictTo('buyer', 'admin'),
+	stripeSuccess
+);
+paymentRoutes.get(
+	'/cancel',
+	// protectRoute,
+	// restrictTo('buyer', 'admin'),
+	stripeCancel
+);
+paymentRoutes.post('/webhook', express.raw({type: 'application/json'}), webHook
```
Comment (ncutixavier) [inline src/routes/api/payment.routes.js]: Remove unnecessary comments

File: src/controllers/payment.controller.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+          };
+          return await SalesService.createSales(sale);
+        })
+      );
+
+      return res.status(200).json({
+        message: 'Order created successfully',
+      });
+    } catch (error) {
+      return res.status(500).json({
```
Comment (ncutixavier) [inline src/controllers/payment.controller.js]: Let's refactor this controller by splitting into different methods and middleware if they are

File: src/controllers/payment.controller.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+			quantity: products[index].quantity,
+			images: product.images,
+		}));
+		const cartTotal = calculateCartTotal(productsDataWithQuantity);
+		const lineItems = productsData.map((product, index) => ({
+			price_data: {
+				currency: 'rwf',
+				unit_amount: product.price,
+				product_data: {
+					name: product.name,
```
Comment (ncutixavier) [inline src/controllers/payment.controller.js]: Let's refactor this controller by splitting it into different methods and middleware if they are

**Review by ncutixavier — CHANGES_REQUESTED** [review]

File: src/utils/payment.utl.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import Stripe from 'stripe';
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY);
+export const createStripeSession = async(lineItems, customer) =>{
+  return await stripe.checkout.sessions.create({
+			payment_method_types: ['card'],
+			shipping_address_collection: {
+				allowed_countries: ['RW', 'UG'],
+			},
+			shipping_options: [
+				{
```
Comment (ncutixavier) [inline src/utils/payment.utl.js]: change filename

**Review by Fidela1 — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**